src/ImageSharp.Web/Providers/IImageProvider.cs

```diff
@@ -14,6 +14,12 @@ namespace SixLabors.ImageSharp.Web.Providers
     /// </summary>
     public interface IImageProvider
     {
+        /// <summary>
+        /// Gets or sets a value indicating whether this provider should process
+        /// an image when no commands have been supplied.
+        /// </summary>
+        bool ProcessWhenNoCommands { get; set; }
```

Comment:
This shouldn't be a settable property on the interface. We don't want it to be configurable during startup

I think we should rename the property also and make it an enum.

```c#
ProcessingBehavior  ProcessingBehavior { get; }
```

``` c#
public enum ProcessingBehavior
{
    CommandOnly,
    All
}
```